Fresh "How to contribute" page (#9609)
godotengine/godot-docs: contributing/how_to_contribute.rst

.. _doc_ways_to_contribute:

How to contribute

> ==================

Reviewer: ```suggestion
=================
```

The Godot Engine is free and open-source. Like any community-driven project, we rely on volunteer contributions. 
On this page we want to showcase the various ways you as users can participate - to help you find the right starting place with the skillset you have.
Because contrary to popular opinion, we need more than just programmers on the project!

Fundraising

> ------------

Reviewer: ```suggestion
-----------
```

  Currently, we are intending to hire more core developers, as to cover more ground with full-time specialists that suplement and guide volunteer work.

  `Join the Development Fund <fund.godotengine.org>`

- **Donation Drives**
  Think about your followers on social media, or other communities you are active in.
  Use that reach to remind your social environment that even small contributions can make a difference, especially when done by a great number of people at the same time. 

  Are you a content creator? Consider adding a link to the `Godot Development Fund <fund.godotengine.org>` to your descriptions. 
  If you do live streams, perhaps think about organizing a stream with donation incentives.

.. - **Buy Official Merch**

- **Publish Godot Games.**
  You heard right, simply publishing a game #MadeWithGodot can positively impact the well-being of this project. 
  Your personal success elevates the engine to a viable alternative for other developers, growing the community further. 
  Additionally, it opens the doors for us to approach industry contacts aboout possible cooperations.

Technical contributions

> ------------------------

Reviewer: ```suggestion
-----------------------

```

  While it is recommended to use the stable releases for your projects, you can help us test dev releases, betas, and release candidates
  by opening a copy of your project in them and checking what problems this introduces or maybe even solves. 
  Make sure to have a backup ready, since this can produce irreversible changes.

  Find recent development versions `here <https://godotengine.org/download/preview/>`
  
- **Contribute Engine Code (mainly C++)**
  The engine development is mainly coordinated on our `Contributor RocketChat <https://chat.godotengine.org/>`, 
  so if you are serious about making PRs you should join us there!

  Read more about the **technical submission process**: :ref:`doc_first_steps`

  *Attention: this page is currently under re-contruction, and some teams might be more active than others*
  For each subject area of the engine, there is a corresponding team to coordinate the work. 
  Join the linked chat to get more eyes on your related PR, learn about open todos, or partake in meetings. 
  For some areas, specialists might even be encouraged to step up as maintainer!
  `List of teams <https://godotengine.org/teams/>`

- **Review Code Contributions**

>   All proposals need to be throroughly reviewed before they can be merged into main.

Author: Both correct, just not as used to the terminology yet, so thanks for calling it out!

- **Demo projects (GDScript, C#, and making Assets)**
  We provide new users with `demo projects <https://github.com/godotengine/godot-demo-projects/>` so they can quickly test new features or get familiar with the engine in the first place.
  At industry events, we might even exhibit these demo projects to showcase what Godot can do! 
  Help improve existing projects or supply your own to be added to the pool, and join the `Contributor RocketChat <https://chat.godotengine.org/channel/demo-content>` to talk about it.

- **Documentation**
  The documentation is one of the most essential parts of any tech project, yet the need to document new features and substential often gets overlooked. 
  Join the `Contributor RocketChat <https://chat.godotengine.org/channel/documentation>` to improve the Godot Engine with your technical writing skills.

- **Translations (spoken languages other than English)**
  Are you interested in making the Godot Engine more accessible to non-English speakers? 
  Community-translations are hosted `here <https://hosted.weblate.org/projects/godot-engine/godot/>`

Community support

> ------------------

Reviewer: ```suggestion
-----------------
```

.. _doc_ways_to_contribute:

How to contribute
=================

The Godot Engine is free and open-source. Like any community-driven project, we rely on volunteer contributions. 
On this page we want to showcase the various ways you as users can participate - to help you find the right starting place with the skillset you have.
Because contrary to popular opinion, we need more than just programmers on the project!

Fundraising
-----------

- **Donate**

  We created the non-profit `Godot Foundation <https://godot.foundation/>` to be able to support the Godot Engine in both matters of finance and administration. 
  In action, this means we can hire people to work part-time or full-time on the project. 
  These jobs include engine development as well as related tasks like code reviews, production management, community & marketing efforts, and more.

>   With as little as 5 Euro per month, you can help us keep going strong. 

Reviewer 2: I live in a euro country but I think this could be confusing to users who don't. Maybe I'm overthinking this :thinking:

Author: I believe the Fund displays in EUR but I would have to check

Author: ```suggestion
  With as little as 5 EUR per month, you can help us keep going strong. 
```

Author: @winston-yallow might be worth noting for you

Reviewer 3: For now we only support payments in Euro. In some countries stripe (the payment provider) may offer automatic conversions depending on the selected payment method, but that feels out of scope to mention here.

Reviewer 2: I kinda assumed it was internationalized, but if it isn't, then keep it.

>   Currently, we are intending to hire more core developers, as to cover more ground with full-time specialists that suplement and guide volunteer work.

Reviewer 4: ```suggestion
  Currently, we are intending to hire more core developers, as to cover more ground with full-time specialists that supplement and guide volunteer work.
```

  `Join the Development Fund <fund.godotengine.org>`

- **Donation Drives**
  Think about your followers on social media, or other communities you are active in.
  Use that reach to remind your social environment that even small contributions can make a difference, especially when done by a great number of people at the same time. 

  Are you a content creator? Consider adding a link to the `Godot Development Fund <fund.godotengine.org>` to your descriptions. 
  If you do live streams, perhaps think about organizing a stream with donation incentives.

.. - **Buy Official Merch**

- **Publish Godot Games.**
  You heard right, simply publishing a game #MadeWithGodot can positively impact the well-being of this project. 
  Your personal success elevates the engine to a viable alternative for other developers, growing the community further. 

>   Additionally, it opens the doors for us to approach industry contacts aboout possible cooperations.

Reviewer 4: ```suggestion
  Additionally, it opens the doors for us to approach industry contacts about possible cooperations.
```

Technical contributions
-----------------------

- **Report bugs & other issues**
  As active users of the engine, you are better equipped to identify bugs and other issues than anyone else.

>   To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new?assignees=&labels=&projects=&template=bug_report.yml>` on our GitHub.

Reviewer 2: ```suggestion
  To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new?template=bug_report.yml>` on our GitHub.
```

Also maybe it would be better to link to `https://github.com/godotengine/godot/issues/new` so that the user can select the report category that best fits their needs.

Author: Good call! Will change this

>   To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new?assignees=&labels=&projects=&template=bug_report.yml>` on our GitHub.

Author: ```suggestion
  To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new>` on our GitHub.
```

Reviewer 2: Since we don't link to the form directly, maybe the text should reflect that. Maybe "file a bug report on our GitHub"?

Btw you can edit the files without creating a comment with a suggestion and applying it. Just go to "file changes" and the has the option to edit the file.
![image](https://github.com/user-attachments/assets/8a432019-6a80-452c-96fc-7a73b85e9f33)

>   Make sure to include as much information as possible to ensure these issues can easily be reproduced by others.

Reviewer 5: A core element of a good bug report is a minimal product, that 
A) makes it possible to reproduce the problem
B) provides a baseline for discussions between bug reporter and potential bug fixer

Please evaluate, if this detail should be mentioned on this page.

Author: On the one hand, certain things cannot be repeated enough, on the other this goes too much into detail for an overview imo - and the linked bug report form does mention these things in detail.

Author: The bug report form contains all this - so I don't know why the "how to contribute page" would need to go into more detail than "the more info the better", we could almost remove that sentence entirely even.

- **Test Development Versions**
  While it is recommended to use the stable releases for your projects, you can help us test dev releases, betas, and release candidates
  by opening a copy of your project in them and checking what problems this introduces or maybe even solves. 
  Make sure to have a backup ready, since this can produce irreversible changes.

  Find recent development versions `here <https://godotengine.org/download/preview/>`
  
- **Contribute Engine Code (mainly C++)**
  The engine development is mainly coordinated on our `Contributor RocketChat <https://chat.godotengine.org/>`, 
  so if you are serious about making PRs you should join us there!

  Read more about the **technical submission process**: :ref:`doc_first_steps`

>   *Attention: this page is currently under re-contruction, and some teams might be more active than others*

Reviewer 4: ```suggestion
  *Attention: this page is currently under re-construction, and some teams might be more active than others*
```

  For each subject area of the engine, there is a corresponding team to coordinate the work. 
  Join the linked chat to get more eyes on your related PR, learn about open todos, or partake in meetings. 
  For some areas, specialists might even be encouraged to step up as maintainer!
  `List of teams <https://godotengine.org/teams/>`

- **Review Code Contributions**

>   All pull requests need to be throroughly reviewed before they can be merged into the master branch.

Reviewer 4: ```suggestion
  All pull requests need to be thoroughly reviewed before they can be merged into the master branch.
```

>   All proposals need to be throroughly reviewed before they can be merged into main.

Author: ```suggestion
  All pull requests need to be throroughly reviewed before they can be merged into the master branch.
```

  Help us get a headstart by participating in the code review process.
  .. add a howto/chat to join
  
  Read more about our **style guide and other requirements**: :ref:`doc_pr_review_guidelines`

- **Write Plugins (GDScript, C#, & more)**
  Community addons are not directly included in the core engine download or repository, yet they provide essential quality of life upgrades for your fellow game developers.
  Upload your plugins to the `Godot Asset Library <https://godotengine.org/asset-library/asset>` to make them available to others.

- **Demo projects (GDScript, C#, and making Assets)**
  We provide new users with `demo projects <https://github.com/godotengine/godot-demo-projects/>` so they can quickly test new features or get familiar with the engine in the first place.
  At industry events, we might even exhibit these demo projects to showcase what Godot can do! 
  Help improve existing projects or supply your own to be added to the pool, and join the `Contributor RocketChat <https://chat.godotengine.org/channel/demo-content>` to talk about it.

- **Documentation**

>   The documentation is one of the most essential parts of any tech project, yet the need to document new features and substential often gets overlooked. 

Reviewer 4: not sure what `substantial` stands for in this context.

Author: substantial changes - this is why you dont use VSCode as a text editor, people 😅

>   The documentation is one of the most essential parts of any tech project, yet the need to document new features and substential often gets overlooked. 

Author: ```suggestion
  The documentation is one of the most essential parts of any tech project, yet the need to document new features and substantial changes often gets overlooked. 
```

  Community-translations are hosted `here <https://hosted.weblate.org/projects/godot-engine/godot/>`

Community support
-----------------

- **Call for Moderators**
  With a community of our size, we need people to step up as volunteer moderators in all kinds of places. 
  These teams are organized by the Godot Foundation, but would not function without the dedication of active community members like you.

  Have a look around your favorite community platform and you might come across open application calls. Currently we are looking for:
  *List under construction*

- **Answer tech-support questions**
  With many new people discovering the Godot Engine recently, the need for peer-to-peer tech-support has never been greater. 
  Be it on the `Forum <https://forum.godotengine.org/>`, our `subreddit <https://www.reddit.com/r/godot/>`, or on `Discord <https://discord.gg/bdcfAYM4W9>`, you can always brighten someone's day by helping them get their personal projects back on track.

- **Create tutorials & more**
  How did you get started with the Godot Engine? 
  Chances are you looked for learning materials outside of what the documentation provides.

>   Without content creators covering the game development process, there would not be this big of a community today.

Reviewer 6: There needs to be a newline at the end of this file, it's why the build is failing.

Author: Oh thanks for explaining that!

>   Without content creators covering the game development process, there would not be this big of a community today.

Author: ```suggestion
  Without content creators covering the game development process, there would not be this big of a community today.
  
```

  Are you a content creator? Consider adding a link to the `Godot Development Fund <fund.godotengine.org>`_ to your descriptions. 
  If you do live streams, perhaps think about organizing a stream with donation incentives.

.. - **Buy Official Merch**

- **Publish Godot Games.**
  You heard right, simply publishing a game #MadeWithGodot can positively impact the well-being of this project. 
  Your personal success elevates the engine to a viable alternative for other developers, growing the community further. 
  Additionally, it opens the doors for us to approach industry contacts about possible cooperations.

Technical contributions
-----------------------

- **Report bugs & other issues**
  As active users of the engine, you are better equipped to identify bugs and other issues than anyone else.
  To let us know about your findings, fill out this `bug report form <https://github.com/godotengine/godot/issues/new/choose>`_ on our GitHub.
  Make sure to include as much information as possible to ensure these issues can easily be reproduced by others.

>   If you are interested in helping keep our bug tracker organized, you can even join the `bugsquad <https://chat.godotengine.org/channel/bugsquad>`_!

Reviewer: ```suggestion
  If you are interested in helping to keep our bug tracker organized, you can even join the `bugsquad <https://chat.godotengine.org/channel/bugsquad>`_!
```

Author: I believe language wise the version without "to" sounds smoother, although both are correct.

Community support
-----------------

- **Call for Moderators**
  With a community of our size, we need people to step up as volunteer moderators in all kinds of places. 
  These teams are organized by the Godot Foundation, but would not function without the dedication of active community members like you.

  Have a look around your favorite community platform and you might come across open application calls. Currently we are looking for:
  *List under construction*

- **Answer tech-support questions**
  With many new people discovering the Godot Engine recently, the need for peer-to-peer tech-support has never been greater. 
  Be it on the `Forum <https://forum.godotengine.org/>`_, our `subreddit <https://www.reddit.com/r/godot/>`_, or on `Discord <https://discord.gg/bdcfAYM4W9>`_, you can always brighten someone's day by helping them get their personal projects back on track.

- **Create tutorials & more**
  How did you get started with the Godot Engine? 
  Chances are you looked for learning materials outside of what the documentation provides.
  Without content creators covering the game development process, there would not be this big of a community today.

>   

Reviewer 6: For the newline to work there needs to be nothing on it and you've got two spaces. So after the "." on the previous line just hit enter once so you've got a blank newline and you're good to go. I'm not sure if that's clear or not, you can see how it is on other pages in the docs as an example.

.. _doc_ways_to_contribute:

How to contribute
=================

The Godot Engine is free and open-source. Like any community-driven project, we rely on volunteer contributions. 
On this page we want to showcase the various ways you as users can participate - to help you find the right starting place with the skillset you have.

> Because contrary to popular opinion, we need more than just programmers on the project!

Author: I think this is a style question, and since the intention is actually less to poke fun at ourselves rather than directly adress that myth I keep hearing ("But I am not a programmer, I can't contribute.") so I would prefer to leave it as is.

  `Join the Development Fund <fund.godotengine.org>`_

- **Donation Drives**
  Think about your followers on social media, or other communities you are active in.
  Use that reach to remind your social environment that even small contributions can make a difference, especially when done by a great number of people at the same time. 

  Are you a content creator? Consider adding a link to the `Godot Development Fund <fund.godotengine.org>`_ to your descriptions. 
  If you do live streams, perhaps think about organizing a stream with donation incentives.

.. - **Buy Official Merch**

- **Publish Godot Games.**
  You heard right, simply publishing a game #MadeWithGodot can positively impact the well-being of this project. 
  Your personal success elevates the engine to a viable alternative for other developers, growing the community further. 
  Additionally, it opens the doors for us to approach industry contacts about possible cooperations.

Technical contributions
-----------------------

- **Report bugs & other issues**
  As active users of the engine, you are better equipped to identify bugs and other issues than anyone else.

>   To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new>`_ on our GitHub.

Reviewer 5: Your current link opens a blank bug report template, missing all important fields (like minimal reproduction project, Godot version number, ...). Please change the link:
```suggestion
  To let us know about your findings, fill out `this bug report form <https://github.com/godotengine/godot/issues/new/choose>`_ on our GitHub.
```

  Make sure to include as much information as possible to ensure these issues can easily be reproduced by others.

  If you are interested in helping fix those problems you discovered, you may also join the `bugsquad <https://chat.godotengine.org/channel/bugsquad>`_!

- **Test Development Versions**
  While it is recommended to use the stable releases for your projects, you can help us test dev releases, betas, and release candidates
  by opening a copy of your project in them and checking what problems this introduces or maybe even solves. 
  Make sure to have a backup ready, since this can produce irreversible changes.

>   Find recent development versions `here <https://godotengine.org/download/preview/>`_

Reviewer 4: `how_to_contribute.rst:4:Duplicate explicit target name: "here".`
one duplicate link remaining.

```suggestion
  You can find our recent development versions on our `godot preview builds <https://godotengine.org/download/preview/>`_ download page.
```

or at `Community-translations are hosted here` (line 91)

(yea I'm not good at writing..)
Theoretical, link names should be able to stand on their own, but I don't think this is normally considered in the docs.

Author: Adapted all links now, thank you!

  
- **Contribute Engine Code (mainly C++)**
  The engine development is mainly coordinated on our `Contributor RocketChat <https://chat.godotengine.org/>`_, 
  so if you are serious about making PRs you should join us there!

  Read more about the **technical submission process**: :ref:`doc_first_steps`

  For each subject area of the engine, there is a corresponding team to coordinate the work. 
  Join the linked chat to get more eyes on your related PR, learn about open todos, or partake in meetings. 
  For some areas, specialists might even be encouraged to step up as maintainer!
  `List of teams <https://godotengine.org/teams/>`_

- **Review Code Contributions**
  All pull requests need to be thoroughly reviewed before they can be merged into the master branch.
  Help us get a headstart by participating in the code review process.

>   .. where should devs go to join these efforts?

Author: Thanks, added!
